napalm/eos/eos.py

```diff
@@ -131,58 +135,119 @@ def _process_optional_args(self, optional_args):
+            sh_ver = self._run_commands(["show version"])
+            if EOSVersion(sh_ver[0]["version"]) >= EOSVersion("4.23.0"):
+                self.cli_version = 2
+        else:
+            try:
+                connection = self.transport_class(
+                    host=self.hostname,
+                    username=self.username,
+                    password=self.password,
+                    timeout=self.timeout,
+                    **self.eapi_kwargs
+                )
 
-            if self.device is None:
-                self.device = Node(connection, enablepwd=self.enablepwd)
-            # does not raise an Exception if unusable
+                if self.device is None:
+                    self.device = Node(connection, enablepwd=self.enablepwd)
+                # does not raise an Exception if unusable
 
-            # let's try to determine if we need to use new EOS cli syntax
-            sh_ver = self.device.run_commands(["show version"])
-            cli_version = (
-                2 if EOSVersion(sh_ver[0]["version"]) >= EOSVersion("4.23.0") else 1
-            )
+                # let's try to determine if we need to use new EOS cli syntax
+                sh_ver = self.device.run_commands(["show version"])
+                self.cli_version = (
+                    2 if EOSVersion(sh_ver[0]["version"]) >= EOSVersion("4.23.0") else 1
+                )
 
-            self.device.update_cli_version(cli_version)
-        except ConnectionError as ce:
-            # and this is raised either if device not avaiable
-            # either if HTTP(S) agent is not enabled
-            # show management api http-commands
-            raise ConnectionException(str(ce))
+                self.device.update_cli_version(self.cli_version)
+            except ConnectionError as ce:
+                # and this is raised either if device not avaiable
+                # either if HTTP(S) agent is not enabled
+                # show management api http-commands
+                raise ConnectionException(str(ce))
 
     def close(self):
         """Implementation of NAPALM method close."""
         self.discard_config()
 
     def is_alive(self):
+        if self.transport == "ssh":
+            null = chr(0)
+            if self.device is None:
+                return {"is_alive": False}
+            try:
+                # Try sending ASCII null byte to maintain the connection alive
+                self.device.write_channel(null)
+                return {"is_alive": self.device.remote_conn.transport.is_active()}
+            except (socket.error, EOFError):
+                # If unable to send, we can tell for sure that the connection is unusable
+                return {"is_alive": False}
+
         return {"is_alive": True}  # always true as eAPI is HTTP-based
 
+    def _run_commands(self, commands, **kwargs):
+        if self.transport == "ssh":
+            fn0039_transform = kwargs.pop("fn0039_transform", True)
```

Comment:
We already have an optional argument, `eos_fn0039_config`, whose value you can access it from `self.fn0039_config` - doesn't have to be per-method. Or could you explain what are you trying to do here?

Reply:
So obviously the main goal is to be able to convert commands between the old and new syntax.

Digging into how this was done for EAPI connections, I found inside the `run_commands` method of the `Node` inside the `pyeapi_syntax_wrapper.py` file the above code, and basically copied it to maintain the logic.

I assume your recommendation here is drop the line
```python
fn0039_transform = kwargs.pop("fn0039_transform", True)
```
and replace the following `if` statement with
```python
if self.fn0039_config
```

Happy to do that - it looks like I missed the significance of that setting as I poked around the code.